### PR TITLE
fix(java): owlbot parent version should use existing parent version

### LINKED ISSUE #1014: Java: when adding new pom.xml files for new service versions, the wrong parent version is referenced

### PR DESCRIPTION
**Pull request by chingor13**
Fixes #1014

### CODE REVIEW

File: docker/owlbot/java/src/fix-poms.py [hunk 1]
```diff
@@ -286,8 +286,8 @@ def main():
         existing_modules[parent_artifact_id] = module.Module(
             group_id=group_id,
             artifact_id=parent_artifact_id,
-            version="0.0.1-SNAPSHOT",
-            release_version="0.0.0",
+            version=main_module.version,
```
Comment (bcoe) [inline docker/owlbot/java/src/fix-poms.py:289]: When owl-bot ran was it just using the hard coded `0.0.1-SNAPSHOT` vs., whatever the library it was running against?

File: docker/owlbot/java/tests/new-version/golden/google-cloud-foo-bom/pom.xml [hunk 2]
```diff
@@ -0,0 +1,95 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>com.google.cloud</groupId>
+  <artifactId>google-cloud-foo-bom</artifactId>
+  <version>0.1.1-SNAPSHOT</version><!-- {x-version-update:google-cloud-foo:current} -->
```
Comment (bcoe) [inline docker/owlbot/java/tests/new-version/golden/google-cloud-foo-bom/pom.xml:6]: Is this the value that would have been incorrect.

File: docker/owlbot/java/cloudbuild_test.yaml [hunk 3]
```diff
@@ -32,3 +32,11 @@ steps:
     args: ["diff", "-rw", "input", "golden"]
     dir: docker/owlbot/java/tests/new-client
     waitFor: ["new-client-run"]
+  - name: "gcr.io/$PROJECT_ID/owlbot-java:$SHORT_SHA"
+    dir: docker/owlbot/java/tests/new-version/input
```
Comment (bcoe) [inline docker/owlbot/java/cloudbuild_test.yaml:36]: I like how test cases end up being additional steps on Cloud Build.

File: docker/owlbot/java/tests/new-version/golden/google-cloud-foo-bom/pom.xml (hunk 2 above)
Comment (chingor13) [reply docker/owlbot/java/tests/new-version/golden/google-cloud-foo-bom/pom.xml:6]: Yeah, you can see the isolated failing check on the merge commit.

Comment (chingor13) [reply docker/owlbot/java/tests/new-version/golden/google-cloud-foo-bom/pom.xml:6]: From the log:
```
--- input/proto-google-cloud-foo-v2/pom.xml
+++ golden/proto-google-cloud-foo-v2/pom.xml
@@ -10,7 +10,7 @@
   <parent>
     <groupId>com.google.cloud</groupId>
     <artifactId>google-cloud-foo-parent</artifactId>
-    <version>0.0.1-SNAPSHOT</version><!-- {x-version-update:google-cloud-foo:current} -->
+    <version>0.1.1-SNAPSHOT</version><!-- {x-version-update:google-cloud-foo:current} -->
   </parent>
   <dependencies>
     <dependency>
```

File: docker/owlbot/java/src/fix-poms.py (hunk 1 above)
Comment (chingor13) [reply docker/owlbot/java/src/fix-poms.py:289]: It's loading the list of modules/versions from the `versions.txt` file which does not contain the parent artifact versions (we assume it matches the main `google-cloud-X` artifact)

**Review by bcoe — APPROVED** [review]

**Pull request merged**